core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java

```diff
@@ -115,7 +115,12 @@ public DefaultMessageBuilder(org.mule.runtime.api.message.Message message) {
 
   @Override
   public InternalMessage.Builder payload(Object payload) {
-    this.payload = payload;
+    if (payload instanceof TypedValue<?>) {
```

Comment:
This is bad idea IMO, what if I want to create a payload that as a TypedValue value? We either need to make payload() always receive a `TypedValue` (preferred) or differentiate this with a different method. Why not rename payload() to value(Object) and use payload(TypedValue), that is probably better than payload(Object) and typedPayload(TypedValue)